### PR TITLE
Feat: products add to cart

### PR DESCRIPTION
**Pull request by sonishreyas**
## Change logs
- Added Add to cart feature including
  - cart-reducer
  - cart-context 
  - helper functions
  - redirection of button to "/auth" if user has not logged In

## Deploy Preview
https://deploy-preview-6--ripple-store-app.netlify.app/

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
shreyas your code is clean.i liked the approach of taking button as separate component depending on the logic you are rendering that. :)

File: src/components/ProductListing/ProductListing.jsx [hunk 1]
```diff
@@ -1,16 +1,22 @@
-import { useProducts } from "../../context";
+import { useNavigate } from "react-router";
+import { useCart, useProducts } from "../../context";
+import {Link} from "react-router-dom";
+import {AddToCartBtn, AddToCartBtnRedirect, AddToWishlistBtn, AddToWishlistBtnRedirect, GoToCartBtn } from "./product-card"
+import { presentInArray } from "../../utils";
```
Comment (ravikumar0403) [inline src/components/ProductListing/ProductListing.jsx:5]: Good to see you have used a utility function for this

File: src/components/ProductListing/ProductListing.jsx [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
 
 const ProductListing = () => {
   const {productsData} = useProducts();
-
+  const token = localStorage.getItem("token");
```
Comment (ravikumar0403) [inline src/components/ProductListing/ProductListing.jsx:9]: reading from localstorage is a sideEffect so it should be done in useEffect

File: src/utils/cart-management.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import axios from "axios";
+
+/**
+ * Add product data to cart
+ * @param {*} element
+ * @param {Object} productData Product to be added in cart
+ * @param {string} token encodedToken of user
+ * @param {function} cartDispatch Reducer function
+ */
```
Comment (ravikumar0403) [inline src/utils/cart-management.js:9]: Good use of JSDocs comments

**Review by ravikumar0403 — APPROVED** [review]
Code looks clean and readable. Good work

**Pull request merged**